include/streamable_flatbuffer.h

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#pragma once
+
+#include <iostream>
+#include <memory>
+#include <type_traits>
+
+#include "../fbs/flatbuffers/flatbuffers.h"
+
+namespace torchcraft {
+
+  // In/OutStreamableFlatBuffer
+  // Wrapper classes to Let us read/write Flatbuffer types to/from a stream.
+  //
+  // We need these because Flatbuffers don't have a fixed size;
+  // they just read from their input pointer until complete,
+  // an don't record the number of bytes consumed in the process.
+  //
+  // So you can't just read in a Flatbuffer from a stream, because you need
+  // to read a chunk the stream in advance; and the amount of stream you need
+  // to read is dynamic; so StreamableFlatbufferWrapper reads/writes metadata
+  // (the size of the stored Flatbuffer) to enable streaming.
+  //
+  // A StreamableFlatbufferWrapper serializes to :
+  //  {
+  //    size_t  The flatbuffer's size,
+  //    t       The flatbuffer
+  //  }
+
+  class OutStreamableFlatBuffer {
+

```

Comment:
Tiny style nit: I usually prefer limiting empty lines to code blocks within the same scope, i.e.: 

```c++
/* instead of this: */
if (a) {

   /* large block */
}

/* rather this: */
if (a)
   /* large block */

   /* another large block on the same scope */
}

/* same scope here as if-block, newline ok */
```